### DISCUSSION TITLE
Grain size dependent mechanical behaviour

### DISCUSSION THREAD

**Discussion by kuduadim** [neperfepx/FEPX#102] [Q&A] [answered]
I am a fresher in polycrystalline plasticity modelling. I would like to study the mechanical behaviour of annealed copper specimens. No second phase precipitates are expected in the material which means the material is single phase. The given FIVE different annealing conditions produce microstructures with various mean grain sizes and their distribution. I would like to know whether the underlying crystal plasticity algorithm in FEPX is size-dependent to obtain the mechanical properties at different annealing conditions (with different mean grain sizes and distribution). Neper would be used for tessellation and meshing.

**Comment by mkasemer** [comment, marked as answer]
Currently FEPX does not explicitly consider size dependence, at least in the way in which you describe. There is a size dependent term in the precipitate hardening model, but this is independent of the morphology of the grains, and I assume from your description that you are seeking more of a Hall-Petch style relationship.

My first question would be if you need to explicitly consider size dependence in the model, or if you can capture the observed behavior using the existing model? It would be worth attempting to generate polycrystals of the same domain size, but with relatively larger or smaller grains, depending on the conditions of the material you wish to inspect. Are you able to capture the desired behavior by considering these different samples?

If you are set on using a Hall-Petch style relationship, this can be added to FEPX, but would require a bit of work to get operational. It would be a good first project to contribute to the codebase.

**Reply by kuduadim** [reply]
Thank you very much for the prompt reply. It is clear now. I have generated different microstructures by varying mean grain size (by keeping the domain dimension constant) and performed FEPX. However, it was not sure whether the observed stress-strain relationship is due to the grain size effect. As the algorithm is size-independent, it must not be accurate. Are you in the processes of incorporating Hall-Petch relation into the FEPX?

**Reply by mkasemer** [reply]
Indeed, the observed effect is related to the relative sizes of grains, but is not directly related to an explicit size term.

At this point in time, I have no intent to personally implement direct size dependence via a Hall-Petch relationship or similar. If you would be interested in this, I would be happy to guide the implementation.

**Reply by RunguangLi** [reply]
@mkasemer I'm also very interested in implementing the size dependence in FEPX, which should be essential for simulating gradient/bimodal material deformation. Could you briefly guide a bit about the related reference and which part in the source Fortran code to modify for it?

**Reply by mkasemer** [reply]
@RunguangLi Sure, if you would like to tackle this, please open an issue on the FEPX-dev repo and we can discuss implementation there.

**Reply by sajutabraham** [reply]
I hope @kuduadim has observed a reduction in the mechanical properties (yield strength and modulus) with an increase in grain size which is obvious. @mkasemer If the observed mechanical properties are related to the relative sizes of the grains, what would be the expected result (or a deviation from the present result) if a size-dependent term is implemented? As @RunguangLi mentioned, for a gradient microstructure deformation, such a size-dependent term would be needed. But, if the microstructure is equiaxed and homogeneous but only deals with changes in mean grain size in different virtual specimens, how does the size-dependent term influence the results? A thought, please.

**Reply by mkasemer** [reply]
All, if a size-dependent strength term is implemented, it would be expected that the simulations would predict a stronger correlation between the size of the grains and the overall strength of material - this is for sure. 

The issue I have is that the Hall-Petch relationship describes an _observation_ (i.e., that materials with smaller grains tend to exhibit higher macroscopic yield strength), rather than a _physical phenomenon_. By making the slip system strength of the lattice dependent on the size of the grain, we are forcing the model to give us the macroscopic results that we want, rather than properly introducing the correct physics to attempt to model observed behavior. To me, this is not necessarily an attractive route, as there aren't many compelling physical arguments that state that the lattice in smaller grains would truly be more resistant to slip. It is the obvious choice to be sure, as slip strength certainly influence yield, but the actual physical motivation is tenuous at best.

Regarding your question as to how the model would capture differences in yield sample-to-sample: consider the macroscopic elastic-plastic behavior of polycrystals. For example, in uniaxial tension, deformation continues to be elastic if there is an elastic ligament connecting the two control surfaces, no matter how thin this ligament. At the moment when a "plastic plane" severs the elastic connection between these surfaces is the macroscopic yield point. Indeed, this has been demonstrated, and shown to correspond proportionally to the offset yield (which itself is a somewhat unsatisfying definition of yield as macroscopic plasticity has already begun). See Figures 6 and 8 here: https://doi.org/10.1007/s11661-019-05187-z . If the sample is a single crystal, it is very easy for this "plane" to traverse the boundaries of the samples and sever elastic connection - the crystal nucleates plasticity at some point, and likely most of the sample has a similar stress state that will cause similar plasticity to occur. The path for this "plastic plane" to traverse the boundaries of the sample and sever elastic connection becomes much more arduous as more and more grains are introduced across the boundary of the sample. In this way, the existing model is in some ways able to capture the essence of grain size effects (see: https://doi.org/10.1016/j.jmps.2017.03.013), indeed I think this would also include bi-modal microstructures, though I have not tested myself.

All that said, I am still happy to discuss where the changes to the code must be made and guide your implementation of such a model if it a strong desire to explicitly include such terms.
